narayana/ArjunaJTA/jta/tests/classes/io/narayana/perf/product/ProductComparison.java

```diff
@@ -131,7 +131,7 @@ protected void doWork(ProductInterface product) throws Exception {
             if (errorCount.incrementAndGet() > MAX_ERRORS) {
                 e.printStackTrace();
 
-                throw new BenchmarkException(e);
+                throw new ExecutionException(e);
```

Comment:
What happens if we throw whatever was thrown before, e.g. throw e;

Reply:
Pull passed but I am not sure if this is the correct way.